app/controllers/api/base_controller.rb

```diff
@@ -71,9 +77,10 @@ def parent_scope
         association = resource_class.reflect_on_all_associations.detect {|assoc| assoc.class_name == 'Host::Base'}
       end
       raise "Association not found for #{parent_name}" unless association
-      result_scope = resource_class_join(association, scope)
+      result_scope = resource_class_join(association, scope).reorder(nil)
       # Check that the scope resolves before return
-      result_scope if result_scope.to_a
+      result_scope.any?
```

Comment:
I do not think we need this line, `result_scope` is always returned even if it is `[]`

Reply:
unless the `result_scope` generates invalid sql, which is what we catch below. it might not be needed after the 5.2 upgrade as that improved scope merging, but for now we need to materialize the query once to make sure there is no error thrown. Since the scope might include millions of records, calling `.to_a` can be very heavy on memory and DB, while `.any?` returns practically immediately even for very complex queries (as it does not actually load any data from the db, just checks if there is something matching - which we don't care anyways).

Reply:
Makes sense.